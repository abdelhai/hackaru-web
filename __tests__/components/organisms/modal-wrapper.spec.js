import Factory from '@/__tests__/__setups__/factory';
import ModalWrapper from '@/components/organisms/modal-wrapper';

describe('ModalWrapper', () => {
  let factory;
  let wrapper;

  beforeEach(() => {
    factory = new Factory(ModalWrapper, {
      stubs: ['modal'],
      propsData: {
        name: 'example'
      },
      slots: {
        default: '<p>Content</p>'
      }
    });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  describe('when emit before-open', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.find({ ref: 'modal' }).vm.$emit('before-open');
    });

    it('emit before-open', () => {
      expect(wrapper.emitted('before-open')).toBeTruthy();
    });
  });
});
