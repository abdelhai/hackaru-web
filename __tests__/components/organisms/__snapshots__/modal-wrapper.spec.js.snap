// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWrapper render correctly 1`] = `
<modal-stub
  adaptive="true"
  height="auto"
  max-width="500"
  name="example"
  pivot-y="0.5"
  transition="fade"
  width="95%"
>
  <div
    class="modal-content"
  >
    <p>
      Content
    </p>
  </div>
</modal-stub>
`;
